### PR TITLE
Project 19 - Petra Almgren

### PR DESCRIPTION
**Pull request by petrame**
https://project-happy-thoughts.netlify.app/
https://happythoughts-api.herokuapp.com/

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,29 +1,95 @@
...
+mongoose.Promise = Promise;
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
+const Message = new mongoose.model("Message", {
+  text: {
+    type: String,
+    required: true,
+    minlength: 5,
+    maxlength: 140,
+  },
```
Comment (KarolCislowski) [inline server.js:16]: nice data structure 👍🏻

File: server.js [hunk 2]
```diff
@@ -1,29 +1,95 @@
...
+const app = express();
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+app.use(cors());
+app.use(bodyParser.json());
+
+// Retrieve the first 20 messages, sorted in descending order
+app.get("/", async (req, res) => {
```
Comment (KarolCislowski) [inline server.js:39]: `/thoughts`

File: server.js [hunk 3]
```diff
@@ -1,29 +1,95 @@
...
+    res.json(messages);
+  } catch {
+    res.status(400).json({
+      message: "No happy thoughts were found 😢",
+    });
+  }
+});
+
+// Post a new happy thought
+app.post("/", async (req, res) => {
```
Comment (KarolCislowski) [inline server.js:54]: `/thoughts`

File: server.js [hunk 4]
```diff
@@ -1,29 +1,95 @@
...
+    // No success
+    res.status(400).json({
+      message: "Couldn't save your happy thought, please hold on to it.",
+      error: err.errors,
+    });
+  }
+});
+
+// Update like button
+app.post("/:messageId/like", async (req, res) => {
```
Comment (KarolCislowski) [inline server.js:75]: `/thoughts/:thoughtId/like`

**Review by KarolCislowski — CHANGES_REQUESTED** [review]
This is a very nice code and your frontend is like from a fairytale! 🌈 🦄  Please change your endpoints paths to meet the requirements 😺

File: server.js [hunk 5]
```diff
@@ -1,29 +1,95 @@
...
+    // Use our moongose model to create the database entry
+    const newMessage = new Message({ text });
+
+    // Success
+    const savedMessage = await newMessage.save();
+    res.status(201).json(savedMessage);
+  } catch (err) {
+    // No success
+    res.status(400).json({
+      message: "Couldn't save your happy thought, please hold on to it.",
```
Comment (emma-tech) [inline server.js:68]: Haha I love the error messages! :) Nice that you made them more fun 🙌

File: server.js [hunk 6]
```diff
@@ -1,29 +1,95 @@
...
+// Update like button
+app.post("/:messageId/like", async (req, res) => {
+  const { messageId } = req.params;
 
-// Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+  try {
+    await Message.updateOne(
```
Comment (emma-tech) [inline server.js:79]: Good that you remembered about updateOne here 🙌

File: server.js [hunk 7]
```diff
@@ -1,29 +1,95 @@
...
+    res.status(400).json({
+      message: "No happy thoughts were found 😢",
+    });
+  }
+});
+
+// Post a new happy thought
+app.post("/", async (req, res) => {
+  try {
+    // Retrieve the information sent by the client to our API endpoint
```
Comment (emma-tech) [inline server.js:56]: Good that you comment your code! Makes it easy to follow :)

File: server.js [hunk 8]
```diff
@@ -1,29 +1,95 @@
...
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+app.use(cors());
+app.use(bodyParser.json());
+
+// Retrieve the first 20 messages, sorted in descending order
+app.get("/", async (req, res) => {
+  const messages = await Message.find()
```
Comment (emma-tech) [inline server.js:40]: Nice and clean code 👍

**Review by emma-tech — COMMENTED** [review]
Good job with the happy thoughts project Petra! Nice, easy to read and well structured code. 
(I can see Karol already left you some comments/changes requested and that you're aware of how to fix those endpoints 🙌)
I also love your frontend, it's so dreamy! 😍🌈

**Review by emma-tech — CHANGES_REQUESTED** [review]
Good job with the happy thoughts project Petra! Nice, easy to read and well structured code.
(I can see Karol already left you some comments/changes requested and that you're aware of how to fix those endpoints 🙌)
I also love your frontend, it's so dreamy! 😍🌈

**Comment by petrame** [comment]
Endpoints updated

**Pull request closed without merging**